src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -233,6 +234,16 @@ private static ScheduledThreadPoolExecutor initSyncExecutor()
 
     private volatile double crcCheckChance;
 
+    public static Iterable<SSTableReader> selectOnlyBigTableReaders(Iterable<? extends SSTableReader> readers)
```

Comment:
Since we are already making code big-format-specific, why not make this also cast to `BigTableReader` and remove the index summary etc. features from the `SSTableReader` interface?

Reply:
Because we would have to move the implementation to a different file, thus making merging upstream changes difficult. We wanted to avoid that